src/trace.rs

```diff
@@ -0,0 +1,136 @@
+use crate::event::command::{
+    CommandEventHandler,
+    CommandFailedEvent,
+    CommandStartedEvent,
+    CommandSucceededEvent,
+};
+use bson::Bson;
+
+pub(crate) struct CommandTracingEventEmitter {
+    max_document_length_bytes: usize,
+}
+
+impl CommandTracingEventEmitter {
+    pub(crate) fn new(max_document_length_bytes: Option<usize>) -> CommandTracingEventEmitter {
+        CommandTracingEventEmitter {
+            max_document_length_bytes: max_document_length_bytes.unwrap_or(1000),
```

Comment:
Nit - I think it'd be a little better to have the default be a named constant.

Reply:
good point. I've added a constant for this on my current working branch.